### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by limavfabio**
This PR adds the classes Person, Student and Teacher for the School Library CLI application

### CODE REVIEW

File: person_class.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  def of_age?
+    @age >= 18
+  end
+
+  public
+
+  def can_use_services?
+    (@age >= 18 || @parent_permission)
+  end
```
Comment (mikethreels) [inline person_class.rb:22]: - [ ] there is no need to use parentheses around this statement
- [ ] Please use the of_age? method to check if the user can use services.

File: student_class.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  attr_accessor :classroom
+
+  def intialize(classroom)
+    @classroom = classroom
+    super(
+      age,
+      name,
+      parent_permission
+    )
+  end
```
Comment (mikethreels) [inline student_class.rb:13]: - [ ] the initialize method should take age, name and parent_permission as arguments as well.

File: teacher_class.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(specialization)
+    @specialization = specialization
+    super(
+      age,
+      name,
+      parent_permission
+    )
```
Comment (mikethreels) [inline teacher_class.rb:12]: - [ ] the initialize method should take age, name and parent_permission as arguments as well.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Fabio, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by alexander16108 — APPROVED** [review]
Hi @limavfabio and @mikethreels 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**